### PR TITLE
Improve GitHub actions

### PR DESCRIPTION
**Pull request by julianprester**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- colrev update runs daily
- colrev update action always pushes to the same branch
- tests do not yet run on different platforms

Issue Number: N/A

## What is the new behavior?

- colrev update runs weekly
- colrev pushes to a unique branch based on uuid
- tests use the matrix testing strategy

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -6,36 +6,42 @@ on:
...
 
   test-full-deps:
     needs: test-minimal-deps
-    runs-on: ubuntu-latest
     strategy:
       matrix:
         platform: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ['3.8', '3.9', '3.10', '3.11']
-
+    runs-on: ${{ matrix.platform }}
```
Comment (julianprester) [inline .github/workflows/tests.yml:34]: I realised that even without this tests seem to run on all platforms. Not sure why... Might consider removing this again.

**Comment by julianprester** [comment]
How are we currently publishing docs to https://colrev.readthedocs.io/en/latest/ ? May be another workflow that could be added in this branch?

**Comment by geritwagner** [comment]
> How are we currently publishing docs to https://colrev.readthedocs.io/en/latest/ ? May be another workflow that could be added in this branch?

Currently, readthedocs automatically pulls content from the main branch ([build](https://readthedocs.org/projects/colrev/builds/)). Are there any advantages if we change it to Github actions? Maybe, an automated docs-build/update or the docs/make linkcheck could be useful.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (geritwagner) [reply .github/workflows/tests.yml:34]: Good point. Could we move the matrix-testing to a new pull-request (and merge this one)?

**Pull request merged**